kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/ssl/SslManagerServiceImpl.java

```diff
@@ -240,17 +239,14 @@ public SSLSocketFactory getSSLSocketFactory(String protocol, String ciphers, Str
     @Override
     public void installTrustCertificate(String alias, X509Certificate x509crt)
             throws GeneralSecurityException, IOException {
-        InputStream tsReadStream = null;
-        FileOutputStream tsOutStream = null;
-
-        try {
+        String trustStore = this.options.getSslKeyStore();
+        File fTrustStore = new File(trustStore);
+        try (InputStream tsReadStream = new FileInputStream(trustStore);
+                FileOutputStream tsOutStream = new FileOutputStream(trustStore)) {
```

Comment:
It seems not to be a good idea to open the same file for reading and writing at the same time. If you move the output stream back down, it should work. You can add another try-with-resources, there.

Reply:
Yes, I've already rolled this back on Friday w/o committing since I did not understand what exactly happens. Thanks!